src/app/SharpRaven/Data/SentryRequestFactory.cs

```diff
@@ -35,7 +35,9 @@
 using System.Linq;
 using System.Reflection;
 using System.Text;
-
+#if net35
+using System.Web;
```

Comment:
The problem lies in the fact that the returned object will be an HttpContext, as defined in System.Web. When moving upwards, you are using dynamics, but that is one of the reasons the original project could not be compiled in .Net 3.5